### PR TITLE
Updated project and schemes for Xcode 13.3

### PR DESCRIPTION
**Pull request by NachoSoto**
The `Package.resolved` file is not backwards compatible, so we shouldn't merge this until we're ready to drop support for `Xcode 13.2.x`.

Also fixed invalid Nimble usages of `===` that were no longer valid since https://github.com/Quick/Nimble/pull/955.

## TODO:
 - [x] Fix test errors with updated Nimble (will do after #1365 is merged).

### CODE REVIEW

**Comment by aboedo** [comment]
do we need to keep Package.resolved versioned, though? I'm not sure we gain much from it, assuming that our dependencies actually stick to semver

**Comment by NachoSoto** [comment]
What do you mean by versioned? I guess we could git ignore it?

**Comment by aboedo** [comment]
> What do you mean by versioned? I guess we could git ignore it?
yep, I meant "under version control"

**Comment by NachoSoto** [comment]
Yeah we only use SPM for test dependencies, so I think that would be totally fine :)

**Comment by NachoSoto** [comment]
@aboedo done.

**Comment by NachoSoto** [comment]
This should be ready now.

File: PurchasesTests/Networking/Backend/BackendCreateAliasTests.swift [hunk 1]
```diff
@@ -156,7 +156,7 @@ class BackendCreateAliasTests: BaseBackendTests {
         })
 
         expect(receivedError).toEventuallyNot(beNil())
-        expect(receivedError?.code).toEventually(be(ErrorCode.invalidCredentialsError.rawValue))
+        expect(receivedError?.code) == ErrorCode.invalidCredentialsError.rawValue
```
Comment (aboedo) [inline PurchasesTests/Networking/Backend/BackendCreateAliasTests.swift:159]: 👌

File: PurchasesTests/Networking/Backend/BackendGetCustomerInfoTests.swift [hunk 2]
```diff
@@ -117,7 +117,7 @@ class BackendGetCustomerInfoTests: BaseBackendTests {
         let underlyingError = (error?.userInfo[NSUnderlyingErrorKey]) as? NSError
         expect(underlyingError).toEventuallyNot(beNil())
         expect(underlyingError?.domain).to(equal("RevenueCat.BackendErrorCode"))
-        expect(error?.userInfo["finishable"]).to(be(true))
+        expect(error?.userInfo["finishable"] as? Bool) == true
```
Comment (aboedo) [inline PurchasesTests/Networking/Backend/BackendGetCustomerInfoTests.swift:120]: out of curiosity, does `beTrue()` work?

**Review by aboedo — APPROVED** [review]

File: PurchasesTests/Networking/Backend/BackendGetCustomerInfoTests.swift (hunk 2 above)
Comment (NachoSoto) [reply PurchasesTests/Networking/Backend/BackendGetCustomerInfoTests.swift:120]: Yeah let me try that for this, should be simpler.

Comment (NachoSoto) [reply PurchasesTests/Networking/Backend/BackendGetCustomerInfoTests.swift:120]: Nope that would also require the cast:
> cannot convert value of type 'Any?' to expected argument type '() throws -> Bool?'

**Pull request merged**